exporter/awsemfexporter/grouped_metric.go

```diff
@@ -89,6 +84,11 @@ func addToGroupedMetric(pmd *pdata.Metric, groupedMetrics map[interface{}]*Group
 	}
 }
 
+func groupedMetricKey(namespace, logGroup, logStream string, timestampMs int64, labels map[string]string) aws.Key {
+	metricName := fmt.Sprintf("%s,%s,%s,%s", namespace, logGroup, logStream, strconv.FormatInt(timestampMs, 10))
```

Comment:
I think you should be able to use a struct instead of a string?

Reply:
Not sure if I understand the question correctly, `metricName` generated from line 88 will be passed to `aws.NewKey`: 
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2627/files#diff-854e1eac3433a8698c435c7df1e08466945f9f0b18a9504ef1aabbd4aee63cb6R89-R92

where `Key` accepts `MetricName` as string (string is used for the purpose of hiding unnecessary business concepts).  Does it answer your question? Or you'd prefer the metricKey to be declared in another way?

Reply:
Yeah I think I'd expect the metric key to be declared as a struct if it's just as for a map - if we need to update the shared library first we could do that. If there's a specific reason to be a string though, other than just being how it's defined right now, then it's fine though.

Reply:
Get it. Done.